MichaelKintscher/qualitative-coding-app
# Refactors and reimplements the scrubber bar

Before this patch, the scrubber bar was a QSlider placed in a QGraphicsScene and viewed in a QGraphicsView. Upon scaling the scrubber bar, the view would zoom in to the respective portion of the slider bar. While this did provide a scaling effect, it had some undesirable side effects. The most problematic was the handle width would zoom in addition to the bar. And, with the handle width becoming an unknown unretrievable quantity, it has become impossible to quantitate the exact width of the slider bar, a necessary factor in computing tick-marks.

This patch reimplements the scrubber bar as a derived QSlider, which dynamically changes its range according to the scaling bar in order to scale, and which hides its handle using an overriden paintEvent method. Now, scaling should be possible, and in addition, the handle width is constant.

This patch also refactors the scalable scrubbing bar files, moving all related files into a related folder for better organization.

Testing Steps:
  1. Run the application
  2. Verify the scrubber bar handle can be moved within its current range
  3. Verify that the progress rect (the blue rectangle spanning from the beginning of the slider to the handle) moves with the handle appropriately.
  4. Load a video
  5. Verify that the scrubber bar works as expected when the scaling bar is set in its default state. 5a. Verify the scrubber bar updates with the video 5b. Verify updates in the scrubber bar update the video
  6. Change the scaling bar value and verify that the range of the scrubber bar accurately represents the scaling bar interval
  7. Verify that the handle is hidden when the progress bar handle represents a value outside the range of the scrubber bar. 7a. When the progress bar handle is less than the minimum of the scrubber bar, the handle and progress rect should not be visible 7b. When the progress bar handle is greater than the maximum of the scrubber bar, the handle should not be visible and the progress rect should occupy the entire width of the scrubber bar.

Type: New Feature
